src/sql/parts/connection/common/connectionManagementService.ts

```diff
@@ -1349,8 +1349,12 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 	 * TODO this could be a map reduce operation
 	 */
 	public buildConnectionInfo(connectionString: string, provider: string): Thenable<sqlops.ConnectionInfo> {
-		return this._providers.get(provider).onReady.then(e => {
-			return e.buildConnectionInfo(connectionString);
-		});
+		let connectionProvider = this._providers.get(provider);
```

Comment:
Note: this was the fix for failure when an extension (in this case, SQL Server 2019 Preview) provides a new connection type, but it's then uninstalled. Without this tests crashed. It should also help avoid unexpected errors during regular product use since this was an unhandled exception.

Reply:
Also I verified that the calling code knows to expect `undefined` and handle that correctly.